aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -97,6 +113,51 @@ module aptos_framework::coin {
     /// Capability required to burn coins.
     struct BurnCapability<phantom CoinType> has copy, store { }
 
+    //
+    // Supply functions
+    //
+
+    /// Creates new supply which can be an aggregator (parallelizable) or a standard
+    /// integer.
+    fun initialize_supply(parallelizable: bool): Supply {
+        Supply {
+            inner: optional_aggregator::new(MAX_U128, parallelizable),
+        }
+    }
+
+    /// Adds `amount` to total supply of `CoinType`. Called when minting coins.
+    fun add(supply: &mut Supply, amount: u128) {
+        optional_aggregator::add(&mut supply.inner, amount);
+    }
+
+    /// Subtracts `amount` from total supply of `CoinType`. Called when burning coins.
+    fun sub(supply: &mut Supply, amount: u128) {
+        optional_aggregator::sub(&mut supply.inner, amount);
+    }
+
+    /// Returns the total supply of `CoinType` in existence.
+    fun read(supply: &Supply): u128 {
+        optional_aggregator::read(&supply.inner)
+    }
+
+    //
+    // Supply config
+    //
+
+    /// Publishes supply configuration. Initially, upgrading is not allowed.
+    public(friend) fun initialize_supply_config(account: &signer) {
+        system_addresses::assert_aptos_framework(account);
+        move_to(account, SupplyConfig { allow_upgrades: false });
+    }
+
+    /// This should be called by on-chain governance to update the config and allow
+    // `Supply` upgradability.
+    public fun allow_supply_upgrades(account: &signer) acquires SupplyConfig {
```

Comment:
Can we make this flexible in case we need to turn it off for whatever reasons? set_allow_supply_upgrades(account: ..., allowed: bool) should work